Automattic/wp-calypso
# New Page Layout Picker: rename "templates" to "page patterns"

#### Changes proposed in this Pull Request

* Refactor JS so the "templates" concept is instead called "patterns" or "page patterns" in the code
* Leave PHP and API as is. We will need to change this more carefully so that it can be deployed without fatals (given the deployment process for ETK has changed recently)
* Update e2e page objects with new CSS selectors

Initially proposed here (https://github.com/Automattic/wp-calypso/pull/49661#pullrequestreview-587067567) when we refactored the modal into it's own package. The word "template" has become very overloaded. The _things_ chosen by the new page modal are really more like "patterns" (i.e. the pattern inserter in core Gutenberg), except they're for a whole page. Hence "Page Patterns".

I also considered calling the package `@automattic/new-page-modal`, since it's the modal that appears when creating a new page. But that's just what we happen to use it for now. Since it's a package it could be used anywhere one needed to select a page pattern.

FYI @noahtallen 

#### Testing instructions



* e2e tests pass (selectors have changed)
* apply diff to sandbox and smoke test the feature still works
* verify that no changes have been made to the API (i.e. the `starterPageTemplatesConfig` global) and that it won't cause fatals when deploying



Related to #50363
